.github/workflows/prepare-release.yml

```
@@ -0,0 +1,35 @@
+name: Update version on new release branch
+
+on:
+  create:
+
+jobs:
+  update_version:
+    if: github.ref_type == 'branch' && startsWith(github.ref, 'refs/heads/release-v')
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+
+    - name: Extract current version
+      id: current-version
+      run: |
+        CURRENT_VERSION=$(grep '^version = ' Scarb.toml | sed 's/version = "\(.*\)"/\1/')
+        echo "CURRENT_VERSION=$CURRENT_VERSION" >> $GITHUB_ENV
+
+    - name: Extract new version number
+      id: new-version
+      run: echo "NEW_VERSION=${GITHUB_REF#refs/heads/release-v}" >> $GITHUB_ENV
+
+    - name: Replace version in files
+      run: |
+        echo "Current version: ${{ env.CURRENT_VERSION }}"
+        echo "New version: ${{ env.NEW_VERSION }}"
+        find . -type f -not -path '*/\.*' -not -path './CHANGELOG.md' -not -path './docs/package-lock.json' -not -path './RELEASING.md' -exec sed -i "s/${{ env.CURRENT_VERSION }}/${{ env.NEW_VERSION }}/g" {} +
```

I'd suggest accessing the environment variables directly. This should work:

```suggestion
      run: |
        echo "Current version: $CURRENT_VERSION"
        echo "New version: $NEW_VERSION"
        find . -type f -not -path '*/\.*' -not -path './CHANGELOG.md' -not -path './docs/package-lock.json' -not -path './RELEASING.md' -exec sed -i "s/$CURRENT_VERSION/$NEW_VERSION/g" {} +
```

This is 100% safe from injections and more native to bash scripts. Though you're right in that the data comes through a more or less trusted source and the current syntax should be fine with the current workflow.

---

thanks!